src/BookStore/Infrastructure/Doctrine/DoctrineBookRepository.php

```diff
@@ -28,13 +28,11 @@ public function __construct(EntityManagerInterface $em)
     public function save(Book $book): void
```

Comment:
With the implemented modifications, it makes more sense now to rename this method to `add()`, in line with the collection-oriented Repository pattern, which is a better fit compared to the persistence-oriented pattern for this demo. 

This is particularly relevant because this demo leverages the Doctrine Unit of Work, so tracking modifications in Aggregates is handled by Doctrine, making the `add()` method a suitable choice.

Reply:
While I agree, I'd like to keep the domain & app layer doctrine-agnostic as it's one of the main goals for this demo

Reply:
I think it's already doctrine-agnostic, as long as you use the port `BookRepositoryInterface` in the app layer. The suggestion is more about the repository pattern I can use: collection-oriented or persistence-oriented?

As repositories are not DAOs in DDD but only represent collections, imho the collection-oriented pattern is a better fit and a simpler concept for this demo.